ACE/ace/OS_NS_sys_utsname.cpp

```diff
@@ -213,17 +213,35 @@ ACE_OS::uname (ACE_utsname *name)
   ACE_OS::strcpy (name->machine, sysModel ());
 
   return ACE_OS::hostname (name->nodename, maxnamelen);
-#elif defined (INTEGRITY)
-  if(!name) {
-    errno = EFAULT;
+#elif defined (ACE_INTEGRITY)
+#  if defined (ACE_LACKS_GETHOSTNAME)
```

Comment:
ACE_LACKS_GETHOSTNAME is checked already in ACE_OS::hostname, so why not call this ACE_OS::hostname and when it returns -1, return -1 in this function